fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/mixin/client/renderer/registry/MixinItem.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.renderer.registry;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.minecraft.item.Item;
+
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.CooldownOverlayProperties;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.CountLabelProperties;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.PostItemOverlayRenderer;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.PreItemOverlayRenderer;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.DurabilityBarProperties;
+import net.fabricmc.fabric.impl.client.renderer.registry.item.ItemOverlayExtensions;
+
+@Mixin(Item.class)
+public abstract class MixinItem implements ItemOverlayExtensions {
```

Comment:
Not a bit fan of doing this, mojang are moving client stuff out of Block/Item (Think render layers). And I think client/common code should have clear boundaries.

Doing this many hashmap lookups might become a peformace issue?

Reply:
it's looking up a `HashMap` *every frame* an item stack is in your hotbar/open screen and caching won't really help here since what's the chance *all stacks on the screen* are of the same item

Reply:
but I've never actually done my own benchmarking, so
I'll just shift this over to use a few `Object2ObjectOpenHashMap`s on the API size

Reply:
https://github.com/FabricMC/fabric/pull/881#discussion_r452966900 apparently you should blame i5 for this "store renderer stuff directly on the `Item` instance" design

Reply:
Moved to `IdentityHashMap`s, which compare equality via `==` instead of `equals`. *Should* have little-to-no impact on performance.